### ISSUE TITLE
Tricky error with unescaped quotes in values

### ISSUE THREAD

**Issue by rockerBOO** [jdidion/pest-test#1]
Hello, thanks for this software. Really helps complete the pest cycle, and I have experience with tree-sitter tests, so it really is nice. So far has worked great.

Ran into this issue which gave odd errors.

```rust
Error: Parser { source: Pest { source: Error { variant: ParsingError { positives: [skip], negatives: [] }, location: Pos(1317), line_col: Pos((65, 12)), path: None, line: "      (json: \"{\"vae\": \"c6a580b13a\", \"model\": \"4d91c4c217\"}\")", continued_line: None } } }
```

Which was suggested from a failing test:

![2023-06-05_23-20](https://github.com/jdidion/pest-test/assets/15027/1aac3c4c-8b6a-4471-810e-15bbd5616663)

Figured it must be something with the quotes after a while, and escaped the quotes, and it works.

[prompt.txt](https://github.com/jdidion/pest-test/files/11659584/prompt.txt)

Alternatively, I thought I didn't care for matching the value with `(json)` vs with the value `(json: "(\"model\"....")` but it said it was missing the value.

No worries as I can work around it. Thanks again.

**Comment by jdidion** [comment]
Yes, that makes sense. Values are represented as strings, so if they are quoted then they need to be escaped. An improvement could be to support using either single- or double-quotes. That way you could avoid escaping by enclosing in single quotes:

```
...
(params_pair
    (key: "Hashes")
    (value
      (json: '{"vae": "c6a580b13a", "model": "4d91c4c217"}')
    )
  )
```

I guess the error message should also re-escape the value.
